app/src/main/java/org/simple/clinic/util/AppSignature.kt

```diff
@@ -0,0 +1,51 @@
+package org.simple.clinic.util
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.pm.PackageManager
+import android.util.Base64
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.util.Arrays
+
+private const val HASH_TYPE = "SHA-256"
+private const val NUM_HASHED_BYTES = 9
+private const val NUM_BASE64_CHAR = 11
+
+@SuppressLint("PackageManagerGetSignatures")
+/**
+ * This is a helper class to generate your message hash to be included in the SMS message.
+ *
+ * Without the correct hash, the app won't receive the message callback. This only needs to be
+ * generated once per app and stored.
+ */
+class AppSignature(private val context: Context) {
+
+  // Get all package signatures for the current package
+  // For each signature create a compatible hash
+  val appSignatures by lazy {
+    val packageName = context.packageName
+    val packageManager = context.packageManager
+
+    val signatures = packageManager.getPackageInfo(packageName,
+        PackageManager.GET_SIGNATURES).signatures
```

Comment:
Styling: when breaking functions, can you either keep all arguments in one line or in separate lines?

Reply:
Can we also move this class to debug sourceset since you're already going to move the usage to DebugClinicApp?